maven-embedder/src/main/java/org/apache/maven/cli/event/ExecutionEventLogger.java

```diff
@@ -53,9 +54,15 @@
 {
     private final Logger logger;
 
-    private static final int LINE_LENGTH = 72;
+    private static final int MAX_LOG_PREFIX_SIZE = 8; // "[ERROR] "
+    private static final int PROJECT_STATUS_SUFFIX_SIZE = 20; // "SUCCESS [  0.000 s]"
+    private static final int MIN_TERMINAL_WIDTH = 60;
+    private static final int MAX_TERMINAL_WIDTH = 130;
+    private static final int TERMINAL_WIDTH =
+            Math.min( MAX_TERMINAL_WIDTH, Math.max( MessageUtils.getTerminalWidth(), MIN_TERMINAL_WIDTH ) );
```

Comment:
I don't think this is a good idea to use a constant for the terminal width.  In the case where the maven class loader is reused, the terminal could change its width.  What about computing this variable and caching it for the `MavenSession` ?

Reply:
If we understand correctly, this would make sure that if a terminal gets resized _during_ a Maven build, the output width would be recomputed accordingly?
If that is the case, we don't understand how storing this value in a `MavenSession` instance rather than a constant would solve this. As far as we know, the `MavenSession` lifetime roughly matches the lifetime of the JVM process, and as such, of the classloader.

Reply:
> If we understand correctly, this would make sure that if a terminal gets resized _during_ a Maven build, the output width would be recomputed accordingly?
> If that is the case, we don't understand how storing this value in a `MavenSession` instance rather than a constant would solve this. As far as we know, the `MavenSession` lifetime roughly matches the lifetime of the JVM process, and as such, of the classloader.

I was thinking about embedded maven in Eclipse and Mvnd, where the maven class loader and classes are reused, but a new session is created for a given build.  I don't think adjusting to the terminal size inside a given build is a good idea.

Reply:
Actually, this use case it not really supported yet.  For this to work, we'd need the information about the terminal width to be set publicly on the MavenSession.  The `MavenCLI` would set it based on a call to `MessageUtils.getTerminalWidth()`.

Reply:
Ah, I didn't realize that tools would create new `MavenSession`'s while reusing the classloaders, thanks for explaining!

I assume the terminal width would have to be publicly available on the session, in order for `mvnd` to create new sessions with the same width. For Eclipse, I assume that they would have to set the width to their internal terminal/logger window.
I'm just not sure whether `MavenSession` is the right class for this, because afaik it's currently cli-unaware. Perhaps there is a better place, but I don't know that by heart.

Reply:
@MartinKanters maybe the `MavenExecutionRequest` would be a better place as it contains already a bunch of things somewhat related to cli output: loggingLevel, showError, etc...

Reply:
Hmm, it's definitely better, but I think it would be best if we could keep this term width state in `maven-embedder`.
What if we store it as a regular field in `ExecutionEventLogger`? Or are you reusing the same instance of that as well in separate runs? Otherwise we could find some way to recalculate it every time `projectDiscoveryStarted()` is called or introduce a new method for it..

Reply:
Then reusing the cache from the `MavenSession` simply to store the value might be the easiest way.  It does not have to leak outside the `ExecutionEventLogger`.